lib/scholar/linear/linear_regression.ex

```diff
@@ -4,12 +4,24 @@ defmodule Scholar.Linear.LinearRegression do
   """
   import Nx.Defn
 
-  @derive {Nx.Container, containers: [:coefficients]}
-  defstruct [:coefficients]
+  @derive {Nx.Container, containers: [:coefficients, :intercept]}
+  defstruct [:coefficients, :intercept]
 
   opts = [
     sample_weights: [
-      type: {:list, {:custom, Scholar.Options, :positive_number, []}}
+      type: {:list, {:custom, Scholar.Options, :positive_number, []}},
+      doc: """
+      The weights for each observation in x. If not provided,
+      all observations are assigned equal weight.
+      """
+    ],
+    fit_intercept?: [
```

Comment:
Perhaps a reference on what the intercept is would be nice

Reply:
Maybe these two sentences would be enough?